support logging of orb-slam2 statistic

apattnaik0721013 opened the pull request:
- #76 is addressed.
- boost::log feature used to provide logging in orb-slam2.
- orb-slam2 entities (frame,local keyframes ,keyframes, local map-points, map-point) are displayed in the log.

apattnaik0721013 commented:
sample log( [orbslam2_garching_3.log](https://github.com/paul-michalik/ORB_SLAM2/files/1980264/orbslam2_garching_3.log) ) is created from garching-3 images.

paul-michalik commented on CMakeLists.txt:
```diff
@@ -53,7 +53,9 @@ add_library(orb-slam2-ext STATIC
-    ext/functionless-sentinel.cpp)
+    ext/functionless-sentinel.cpp
+    ext/statistics.h	
+    ext/statistics.cc)
```
Please use unified extensions for ***.cpp** source files!

paul-michalik commented on ext/statistics.h:
```diff
@@ -0,0 +1,87 @@
+		public:
+			// The constructor initializes the internal data
+			explicit stat_collector(std::string &file_name):m_csv_file(file_name, std::ios::out)
+			{ if (!m_csv_file.is_open()) throw std::runtime_error("could not open the CSV file"); }
```
Please write every statement on separate line!

paul-michalik commented on ext/statistics.h:
```diff
@@ -0,0 +1,87 @@
+namespace ORB_SLAM2
+{
+	namespace ext {
+
```
Please do not alias namespaces like that. I'd prefer no aliasing at all, but if you insist use `namespace log = boost::log;` and be explicit about the others, i.e. `log::sources`

paul-michalik commented on src/System.cc:
```diff
@@ -426,6 +424,18 @@ void System::SaveKeyFrameTrajectoryTUM(const string &filename)
+        }
+    }
+
+    ext::statistics::get().update_status(vpKFs.size(), vpMP.size(), semantic_mappoints);
```
This isn't very concise... If you want to provide a semantic logger API which understands the domain, please factor the entire logic of fetching the map points and dump them into a function, i.e. something like:
```
ext::statistics::get().log_map_state(mpMap);
```

paul-michalik commented on ext/statistics.h:
```diff
@@ -0,0 +1,87 @@
+#include <vector>
+#include <opencv2/core/types.hpp>
+
+#include "Map.h"
```
If you decide to pull in everything from `boost/log` here, why is it necessary to split between header and implementation? Either hide implementation details in an implementation file and expose what is the public logger API here, or leave everything in one file...

apattnaik0721013 commented:
@paul-michalik 

- i have addressed all the review comments.
- sample generated stats file [orbslam.xlsx](https://github.com/paul-michalik/ORB_SLAM2/files/2016864/orbslam.xlsx) is attached.
Note: 
- as i am not able upload csv file hence formatted xlsx file is attached.
- this stats is generated from Garching loop closer 3 images with semantic data.

paul-michalik commented:
> as i am not able upload csv file hence formatted xlsx file is attached.

What do you mean you are unable to upload csv file...? :-)
